store/supabase.js

```diff
@@ -0,0 +1,133 @@
+// state
+export const state = () => ({
+  topics: [],
+  members: [],
+  edges: [],
+  nodes: [],
+});
+
+export const actions = {
+  async fetchData({ commit }) {
+    // hit supabase to get topics, members and mappings
+    const { data: topics, error: topicError } = await this.$supabase
+      .from("topics")
+      .select("*");
+    const { data: members, error: memberError } = await this.$supabase
+      .from("members")
+      .select("*");
+    const { data: mapping, error: mappingError } = await this.$supabase
+      .from("members_topics")
+      .select("*");
+
+    if (!topicError) {
+      commit("setTopics", topics);
+    }
+    if (!memberError) {
+      commit("setMembers", members);
+    }
+
+    if (!topicError && !memberError && !mappingError) {
+      const { edges, nodes } = createNodes(members, topics, mapping);
+      commit("setEdges", edges);
+      commit("setNodes", nodes);
+    }
+  },
+};
+
+export const mutations = {
+  setTopics(state, topics) {
+    state.topics = topics;
+  },
+  setMembers(state, members) {
+    state.members = members;
+  },
+  setEdges(state, edges) {
+    state.edges = edges;
+  },
+  setNodes(state, nodes) {
+    state.nodes = nodes;
+  },
+};
+
+function createNodes(members, topics, mappings) {
+  let nodes = [];
+  let memberNames = {};
+  for (const member of members) {
+    nodes.push({
+      _cssClass: "Member",
+      _labelClass: "memberLabel",
+      name: member.username,
+      id: `member-${member.id}`,
+      submittedAd: member.created_at,
+    });
+    memberNames[member.id] = member.username;
+  }
+
+  for (const topic of topics) {
+    nodes.push({
+      _cssClass: "Skill",
+      _labelClass: "skillLabel",
+      name: topic.name,
+      id: `topic-${topic.id}`,
+      sharerNames: [],
+      learnerNames: [],
+    });
+  }
+
+  const edges = [];
+  const counter = { learners: {}, sharers: {} };
+
+  for (const topic of mappings.filter((item) => item.learner)) {
```

Comment:
It takes me quite some time to wrap my head around how this loop is adding learning relationships to the edges array. Perhaps add a little comment that describes what the function is doing?